buildengine/filehash.go

```diff
@@ -122,3 +126,34 @@ func ComputeFileHash(project Project, srcPath string) (hash []byte, matched bool
 	}
 	return nil, false, nil
 }
+
+// computeRootDirs computes the unique root directories for the given baseDir and patterns.
+func computeRootDirs(baseDir string, patterns []string) []string {
+	uniqueRoots := make(map[string]struct{})
+	uniqueRoots[baseDir] = struct{}{}
+
+	for _, pattern := range patterns {
+		fullPath := filepath.Join(baseDir, pattern)
+
+		// Remove any file-globbing patterns which could include '*' or '**'
+		var dirPath string
```

Comment:
I think there's a function in the glob package we use to return the base directory of the glob.

Reply:
OMG that would be so nice, this felt so overkill when I was writing it